ckanext/datastore/logic/action.py

```diff
@@ -660,8 +660,10 @@ def set_datastore_active_flag(
     # find changed resource, patch it and reindex package
     psi = search.PackageSearchIndex()
     try:
+        resource = model.Resource.get(data_dict['resource_id'])
+        assert resource
         _data_dict = p.toolkit.get_action('package_show')(context, {
-            'id': model.Resource.get(data_dict['resource_id']).package_id
+            'id': resource.package_id
```

Comment:
Clearly this change is unrelated to the pull request intent, but tests were failing...